### PR TITLE
AEROGEAR-3411 - Add push is Registered dialog

### PR DESCRIPTION
**Pull request by ciaranRoche**
## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-3411

## Screenshot

push-not-reg

## Verify

- Edit push url in `mobile-service.json`
- Start App
- Navigate to push page

### CODE REVIEW

File: src/services/push.service.ts [hunk 1]
```diff
@@ -65,7 +66,8 @@ export class PushService {
         PushService.registered = true;
         console.log("Push registration successful");
       }).catch(err => {
-        console.log(err.message);
+        this.errorMessage = err.message;
+        console.log(this.errorMessage);
```
Comment (aidenkeating) [inline src/services/push.service.ts]: We can remove this

File: src/services/push.service.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export class PushService {
   static callback: (notification: PushMessage) => void;
 
   public messages: PushMessage[] = [];
+  private errorMessage: string;
```
Comment (aidenkeating) [inline src/services/push.service.ts]: Could we store the `Error` instead of just the message

**Comment by ciaranRoche** [comment]
@aidenkeating mind having another look.

File: src/app/app.component.ts [hunk 3]
```diff
@@ -86,12 +88,22 @@ export class MyApp {
...
+          this.alert.showAlert(constants.pushAlertMessage, constants.featureNotConfigured,
+            constants.alertButtons, constants.showDocs, constants.pushDocsUrl);
+          return;
+        }
+        // checking if push has registered
+        if (!this.pushService.isRegistered()){
+          this.alert.showAlert(`The push service failed to register. \n\n Details: ${this.pushService.getError().message}`, "Push Not Registered", ["Close"], "")
+          return;
+        }
         return;
```
Comment (aidenkeating) [inline src/app/app.component.ts]: Do we want this return statement? Won't that stop every navigation attempt?

File: src/app/app.component.ts [hunk 4]
```diff
@@ -97,11 +97,11 @@ export class MyApp {
           return;
         }
         // checking if push has registered
-        if (!this.pushService.isRegistered()){
+        if (this.pushService.getError()){
+          console.log("isRegistered");
```
Comment (aidenkeating) [inline src/app/app.component.ts]: We can remove this

**Pull request merged**